main.go

```diff
@@ -74,8 +74,8 @@ func processRequests(deathNote map[string]bool, firstConnected chan<- bool, wg *
 			panic(err)
 		}
 		log.Printf("New client connected: %s\n", conn.RemoteAddr().String())
+		wg.Add(1)
```

Comment:
Could you send this potential fix including explanation in a separate PR? 🙏 

We are going to close this PR as the dependencies will be bumped in #46